src/schema/v2/reverseImageSearch.ts

```diff
@@ -0,0 +1,133 @@
+import { ReadStream } from "fs"
+import {
+  GraphQLFieldConfig,
+  GraphQLFloat,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLScalarType,
+  GraphQLString,
+  GraphQLList,
+} from "graphql"
+import { GraphQLUpload } from "graphql-upload"
+import { ResolverContext } from "types/graphql"
+
+export const ReverseImageSearchResultMatchRect = new GraphQLObjectType({
+  name: "ReverseImageSearchResultMatchRect",
+  fields: () => ({
+    left: {
+      type: new GraphQLNonNull(GraphQLFloat),
+    },
+    right: {
+      type: new GraphQLNonNull(GraphQLFloat),
+    },
+    top: {
+      type: new GraphQLNonNull(GraphQLFloat),
+    },
+    bottom: {
+      type: new GraphQLNonNull(GraphQLFloat),
+    },
+  }),
+})
+
+export const ReverseImageSearchResult = new GraphQLObjectType({
+  name: "ReverseImageSearchResult",
+  fields: () => ({
+    score: {
+      description: "How closely the query image matches the collection image",
+      type: new GraphQLNonNull(GraphQLFloat),
+    },
+    filepath: {
+      description: "The matching collection image’s file path",
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    matchPercent: {
+      description:
+        "How much of the query image’s fingerprint overlaps the collection image’s fingerprint, as a percentage",
+      type: new GraphQLNonNull(GraphQLFloat),
+      resolve: ({ match_percent }) => match_percent,
+    },
+    queryOverlapPercent: {
+      description:
+        "How much of the query image overlaps the collection image, as a percentage",
+      type: new GraphQLNonNull(GraphQLFloat),
+      resolve: ({ query_overlap_percent }) => query_overlap_percent,
+    },
+    targetOverlapPercent: {
+      description:
+        "How much of the collection image overlaps the query image, as a percentage",
+      type: new GraphQLNonNull(GraphQLFloat),
+      resolve: ({ target_overlap_percent }) => target_overlap_percent,
+    },
+    targetMatchRect: {
+      description: "Location of the matching area in the second image",
+      type: new GraphQLNonNull(ReverseImageSearchResultMatchRect),
+      resolve: ({ target_match_rect }) => target_match_rect,
+    },
+    queryMatchRect: {
+      description: "Location of the matching area in the first image",
+      type: new GraphQLNonNull(ReverseImageSearchResultMatchRect),
+      resolve: ({ query_match_rect }) => query_match_rect,
+    },
+  }),
+})
+
+export const ReverseImageSearchResults = new GraphQLObjectType({
+  name: "ReverseImageSearchResults",
+  fields: {
+    results: {
+      type: new GraphQLList(ReverseImageSearchResult),
```

Comment:
#minor should this be nullable (i'd imagine an empty result set would be an empty list vs a `null`)?

Reply:
I don't think that `results` should be nullable. We should return an empty array (as it happens now) if there are no matches

Reply:
(in that case maybe `new GraphQLNonNull(new GraphQLList(...` is the right invocation for the typing)